core/src/ru/spbau/blackout/entities/GameObject.java

```diff
@@ -16,30 +16,60 @@
 import com.badlogic.gdx.utils.Pool;
 
 import ru.spbau.blackout.GameWorld;
+import ru.spbau.blackout.utils.Creator;
+import ru.spbau.blackout.utils.InplaceSerializable;
 
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
 import java.io.Serializable;
 
 import static ru.spbau.blackout.utils.Utils.fixTop;
 
-
-public abstract class GameObject implements RenderableProvider, Serializable {
+public abstract class GameObject implements RenderableProvider, InplaceSerializable, Serializable {
     // physics:
     transient protected Body body;
-    float height;
+    private float height;
 
     // appearance:
     transient protected ModelInstance model;
 
     protected GameObject(Definition def, Model model, GameWorld gameWorld) {
-        this.model = new ModelInstance(model);
+        this.model = model == null ? null : new ModelInstance(model);
 
         body = gameWorld.addObject(this, def);
-        body.createFixture(def.fixtureDef);
+
+        FixtureDef fixtureDef = new FixtureDef();
+        fixtureDef.shape = def.getShapeCreator().create();
+        fixtureDef.density = def.getDensity();
+        fixtureDef.friction = 0;
+        fixtureDef.restitution = 0;
+        body.createFixture(fixtureDef);
+        fixtureDef.shape.dispose();
 
         setPosition(def.getPosition().x, def.getPosition().y);
     }
 
     @Override
+    public void inplaceSerialize(ObjectOutputStream out) throws IOException, ClassNotFoundException {
+        out.writeObject(this);
+        out.writeObject(this.getPosition());
```

Comment:
No, it isn't because position isn't a field of GameObject. It's something from a transient field called "body". That is why we have to implement own serialization interface.